### PR TITLE
add integration tests for  access path selection

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Add two integration tests for access path selection.

### What is changed and how it works?
Add two files `access_path_selection.text` and `access_path_selection.result` to store test SQL and results correspondingly.

### Check List 

Tests 
 - Integration test

Code changes

Side effects

Related changes

### CODE REVIEW

File: cmd/explaintest/t/access_path_selection.test [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+`c_id` bigint(11) DEFAULT NULL,
+`c_name` varchar(50) DEFAULT NULL,
+`d_id` bigint(11) DEFAULT NULL,
+`d_name` varchar(32) DEFAULT NULL,
+`is_custom` varchar(32) DEFAULT NULL,
+`io_type` int(11) DEFAULT NULL,
+`p_cnt` bigint(11) DEFAULT NULL,
+`p_amt` bigint(11) DEFAULT NULL,
+`update_source` int(11) DEFAULT NULL,
+`update_time` timestamp DEFAULT NULL,
```
Comment (lamxTyler) [inline cmd/explaintest/t/access_path_selection.test]: Can we remove the useless columns?

Comment (qw4990) [reply cmd/explaintest/t/access_path_selection.test]: of course.
All useless columns have been removed now.

File: cmd/explaintest/t/access_path_selection.test [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+explain select id from pay_daily where date_key>12565 and a_id<159298363;
+
+CREATE TABLE `access_path_selection` (
+`a` int,
+`b` int,
+KEY `IDX_a` (`a`),
+KEY `IDX_b` (`b`),
+KEY `IDX_ab` (`a`, `b`)
+);
+insert into `access_path_selection` values (1, 0);
```
Comment (winoros) [inline cmd/explaintest/t/access_path_selection.test]: Since it uses pseudo statistics. Do we need to insert values?

Comment (qw4990) [reply cmd/explaintest/t/access_path_selection.test]: OK, these useless insert statements have been removed.

**Comment by qw4990** [comment]
PTAL~ @winoros @lamxTyler

**Comment by qw4990** [comment]
Add a more wrong index selection case.  PTAL~ @lamxTyler

File: cmd/explaintest/r/access_path_selection.result [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+INDEX idx_ab(a,b)
+);
+insert into wrong_index_selection values (1, 1, 1);
+insert into wrong_index_selection values (2, 2, 2);
+insert into wrong_index_selection values (3, 3, 3);
+analyze table wrong_index_selection index idx_ab;
+explain select * from wrong_index_selection where a=1 and b=1 and c=1;
+id	count	task	operator info
+IndexLookUp_11	0.80	root	
+├─IndexScan_8	1.00	cop	table:wrong_index_selection, index:a, range:[1,1], keep order:false
```
Comment (lamxTyler) [inline cmd/explaintest/r/access_path_selection.result]: I think in this case, the estimation row count is right, and it is not a wrong choice.

Comment (qw4990) [reply cmd/explaintest/r/access_path_selection.result]: The case has been updated, how about now?

**Comment by qw4990** [comment]
One more bad case involving unknown correlation between columns has been added.
PTAL~ @lamxTyler @eurekaka

File: cmd/explaintest/r/access_path_selection.result [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+IndexLookUp_18	0.00	root	
+├─IndexScan_16	0.00	cop	table:pay_daily, index:a_id, b_id, date_key, range:[12565 159298363 20180731,12565 159298363 20180731], keep order:false, stats:pseudo
+└─TableScan_17	0.00	cop	table:pay_daily, keep order:false, stats:pseudo
+explain select id from pay_daily where a_id=12565 and b_id=159298363 and time_stamp=10000;
+id	count	task	operator info
+Projection_4	0.00	root	test.pay_daily.id
+└─IndexLookUp_10	0.00	root	
+  ├─IndexScan_8	0.00	cop	table:pay_daily, index:a_id, b_id, time_stamp, range:[12565 159298363 10000,12565 159298363 10000], keep order:false, stats:pseudo
+  └─TableScan_9	0.00	cop	table:pay_daily, keep order:false, stats:pseudo
+explain select id from pay_daily where update_source>12565 and a_id<159298363;
```
Comment (eurekaka) [inline cmd/explaintest/r/access_path_selection.result]: How can we decide if `IDX_US` is better or `IDX_ID`/`IDX_TENANT_POI_DATEKEY ` is better for this query when we don't have stats info?

File: cmd/explaintest/r/access_path_selection.result [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  ├─IndexScan_8	3333.33	cop	table:pay_daily, index:date_key, range:(12565,+inf], keep order:false, stats:pseudo
+  └─TableScan_9	3333.33	cop	table:pay_daily, keep order:false, stats:pseudo
+explain select id from pay_daily where date_key>12565 and io_type=1;
+id	count	task	operator info
+Projection_4	3.33	root	test.pay_daily.id
+└─IndexLookUp_11	3.33	root	
+  ├─IndexScan_8	3333.33	cop	table:pay_daily, index:date_key, range:(12565,+inf], keep order:false, stats:pseudo
+  └─Selection_10	3.33	cop	eq(test.pay_daily.io_type, 1)
+    └─TableScan_9	3333.33	cop	table:pay_daily, keep order:false, stats:pseudo
+explain select id from pay_daily where date_key>12565 and a_id<159298363;
```
Comment (eurekaka) [inline cmd/explaintest/r/access_path_selection.result]: Ditto

File: cmd/explaintest/t/access_path_selection.test [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+);
+insert into outdated_statistics values (2, 2, 2);
+insert into outdated_statistics values (3, 3, 3);
+insert into outdated_statistics values (4, 4, 4);
+analyze table outdated_statistics;
+insert into outdated_statistics values (1, 1, 1);
+insert into outdated_statistics values (1, 2, 2);
+insert into outdated_statistics values (1, 3, 3);
+# only update idx_ab and leave idx_a outdated.
+# then the estimated number of rows on idx_ab is 1 while it's 0 on idx_a.
```
Comment (eurekaka) [inline cmd/explaintest/t/access_path_selection.test]: Please add comment here to inform that `idx_ab` should always be better than `idx_a` for this query, and the explain result would be updated later when skyline pruning is merged?

File: cmd/explaintest/t/access_path_selection.test [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    INDEX idx_a(a)
+);
+INSERT INTO unknown_correlation values (1, 1),(2, 1),(3, 1),(4, 1),(5, 1),(6, 1),(7, 1),(8, 1),(9, 1),(10, 1),(11, 1),(12, 1),(13, 1),(14, 1),(15, 1),(16, 1),(17, 1),(18, 1),(19, 1),(20, 2),(21, 2),(22, 2),(23, 2),(24, 2),(25, 2);
+ANALYZE TABLE unknown_correlation;
+# Estimated row count on idx_a is 6,
+# while the estimated row count on TableScan is 4.17, which is computed as below:
+#   selectivity := ds.stats.RowCount / rowCount   ==> selectivity = 6 / 25
+#   rowCount := prop.ExpectedCnt / selectivity    ==> rowCount = 1 / (6 / 25) = 4.17
+# So the planner will pick TableScan instead of IndexScan to execute.
+# But actually the cost of TableScan is 25-6+1=20 under this correlation between `id` and `a`.
```
Comment (eurekaka) [inline cmd/explaintest/t/access_path_selection.test]: Ditto, add a comment to inform that this is not an optimal plan.

File: cmd/explaintest/r/access_path_selection.result (hunk 4 above)
Comment (qw4990) [reply cmd/explaintest/r/access_path_selection.result]: This unclear case has been removed now.

File: cmd/explaintest/t/access_path_selection.test (hunk 6 above)
Comment (qw4990) [reply cmd/explaintest/t/access_path_selection.test]: More comments have been added, how about now?

**Pull request merged**